tests/domain/test_transient.py

```diff
@@ -30,28 +30,27 @@ class TransientDomain(LibvirtTest):
     def setUp(self):
         super().setUp()
         self.image = self.get_generic_image()
+        self.domain = self.create_domain(arguments={'image': self.image.path})
```

Comment:
If we decide to turn `arguments` into a kwargs, this would become:

```python
self.domain = self.create_domain(image: self.image.path)
```

Which seems much nicer to users.  But, we have to measure the flexibility decrease on `create_domain()` and how often users will interact with `create_domain()`.

Reply:
To be honest `self.create_domain(image: self.image.path)`  doesn't look very pythonic, quite the contrary I'd say.

Reply:
After further consideration and looking properly at the `create_domain` method, using `kwargs` is actually not a bad idea at all - I've seen similar handling in both Linode and IBM cloud SDKs (especially the latter one). The ever-present issue with `kwargs` is that it often lacks documentation because the caller doesn't know where `kwargs` will eventually end up being passed through to. If we decide to use `kwargs` here, apart from docs we'll need to make sure, it will never be used for anything else other than the generic template.

Reply:
Yes, we need to pay attention to the documentation, but If you guys allow me, will change this after first draft is sent.

Reply:
Oh, sure! As far as I'm concerted, my comments were really on the "food for thought" and "future ideas" categories.  Nothing that I would require/expect to definitely have implemented (that's why I approved this PR).